kanidmd/src/lib/be/idl_arc_sqlite.rs

```diff
@@ -646,7 +730,13 @@ impl<'a> IdlArcSqliteWriteTransaction<'a> {
         // Drop the entry cache.
         self.entry_cache.clear();
         // Write the raw ents
-        self.db.write_identries_raw(au, entries)
+        self.db
+            .write_identries_raw(audit, entries)
+            .and_then(|()| self.db.get_allids(audit))
+            .map(|mut ids| {
+                // Update allids since we cleared them and need to reset it in the cache.
+                std::mem::swap(self.allids.deref_mut(), &mut ids);
+            })
```

Comment:
For anyone playing at home - this is the section that fixes the restore process. Everything else is about adding diagnostic tooling, or better logging.